src/core/initialize.rs

```diff
@@ -39,16 +39,23 @@ pub fn initialize(pid: pid_t) -> Result<StackTraceGetter, Error> {
 
 // Use a StackTraceGetter to get stack traces
 pub struct StackTraceGetter {
-    process: Process<ProcessHandle>,
+    process: Process,
     current_thread_addr_location: usize,
-    stack_trace_function: StackTraceFn<ProcessHandle>,
+    stack_trace_function: StackTraceFn,
     reinit_count: u32,
 }
 
 impl StackTraceGetter {
     pub fn get_trace(&mut self) -> Result<StackTrace, Error> {
         match self.get_trace_from_current_thread() {
-            Ok(trace) => return Ok(trace),
+            Ok(mut trace) => return {
+                /* This is a spike to enrich the trace with the pid.
```

Comment:
Is this something that will be moved into remoteprocess later?

Reply:
Getting the trace? No, not entirely. rbspy is still going to fetch traces from Ruby processes. Though, remoteprocess can fetch native stack traces and that's something I would like to use in the future.